File: be-fschool/src/main/java/com/tinder/befschool/entity/Attendance.java
package com.tinder.befschool.entity;

import jakarta.persistence.*;
import java.time.LocalDate;

@Entity
@Table(name = "attendance")
public class Attendance extends Auditable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "student_id", nullable = false)
    private Long studentId;

    @Column(name = "teacher_id")
    private Long teacherId;

    @Column(name = "class_name")
    private String className;

    @Column(nullable = false)
    private LocalDate date;

    // PRESENT, ABSENT, LATE
    @Column(nullable = false)
    private String status;

    private String note;

    private String subject;

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public Long getStudentId() { return studentId; }
    public void setStudentId(Long studentId) { this.studentId = studentId; }

    public Long getTeacherId() { return teacherId; }
    public void setTeacherId(Long teacherId) { this.teacherId = teacherId; }

    public String getClassName() { return className; }
    public void setClassName(String className) { this.className = className; }

    public LocalDate getDate() { return date; }
    public void setDate(LocalDate date) { this.date = date; }

    public String getStatus() { return status; }
    public void setStatus(String status) { this.status = status; }

    public String getNote() { return note; }
    public void setNote(String note) { this.note = note; }

    public String getSubject() { return subject; }
    public void setSubject(String subject) { this.subject = subject; }
}

File: prm/lib/models/attendance.dart
class Attendance {
  final int id;
  final int studentId;
  final int? teacherId;
  final String? className;
  final String date; // yyyy-MM-dd
  final String status; // PRESENT, ABSENT, LATE
  final String? note;
  final String? subject;

  const Attendance({
    required this.id,
    required this.studentId,
    this.teacherId,
    this.className,
    required this.date,
    required this.status,
    this.note,
    this.subject,
  });

  bool get isPresent => status == 'PRESENT';
  bool get isAbsent => status == 'ABSENT';
  bool get isLate => status == 'LATE';

  String get statusLabel {
    switch (status) {
      case 'PRESENT':
        return 'Có mặt';
      case 'ABSENT':
        return 'Vắng';
      case 'LATE':
        return 'Đi muộn';
      default:
        return status;
    }
  }

  factory Attendance.fromJson(Map<String, dynamic> json) {
    return Attendance(
      id: json['id'] ?? 0,
      studentId: json['studentId'] ?? 0,
      teacherId: json['teacherId'],
      className: json['className']?.toString(),
      date: json['date']?.toString() ?? '',
      status: json['status']?.toString() ?? 'PRESENT',
      note: json['note']?.toString(),
      subject: json['subject']?.toString(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'studentId': studentId,
      'teacherId': teacherId,
      'className': className,
      'date': date,
      'status': status,
      'note': note,
      'subject': subject,
    };
  }
}
